Batteries/Data/List/Lemmas.lean

```diff
@@ -736,6 +752,17 @@ theorem findIdx?_of_eq_none {xs : List α} {p : α → Bool} (w : xs.findIdx? p
     simp only [replicate, findIdx?_cons, Nat.zero_add, findIdx?_succ, Nat.zero_lt_succ, true_and]
     split <;> simp_all
 
+theorem findIdx_eq_findIdx? (p : α → Bool) (l : List α) :
+    l.findIdx p = (match l.findIdx? p with | some i => i | none => l.length) := by
```

Comment:
Is this easier to state and prove with Option.elim?

Reply:
Or just `Option.getD`:
```suggestion
    l.findIdx p = (l.findIdx? p).getD l.length := by
```